### DISCUSSION TITLE
restart after binding directories

### DISCUSSION THREAD

**Discussion by kexinliu0001** [nextpyp/.github#26] [Q&A] [answered]
hi,

I bindedd directories as installation step 5 to make files visible to PYP.
but when I tried to restart, it failed.

how can I overcome it?

thx!!

systemctl restart nextPYP:
![WechatIMG127](https://github.com/nextpyp/.github/assets/110724494/26c80783-8dd6-4780-89b0-a40bd16fd28f)

systemctl status nextPYP.service:
![WechatIMG128](https://github.com/nextpyp/.github/assets/110724494/b64caf0d-dfe5-419c-a893-0df49beb0514)

I also tried "journalctl -u nextPYP" as in troubleshooting: 
<img width="1101" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/6d196ac8-8f84-404b-a1bb-fad2ed08c35f">

**Comment by abartesaghi** [comment]
We can't see the attachments, can you upload the files again?

**Reply by kexinliu0001** [reply]
thank you for reminding！I've updated again.

and I also tried not binding directories (without "binds = []" in confit.toml), but the it still cannot be restarted.

I tried to reinstall nextPYP, but it still not work 

PYP_USER=nextpyp ./install
<img width="1149" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/abaf53c3-831a-471c-9133-21106f8d8cb3">

**Reply by abartesaghi** [reply]
As suggested in the [installation docs](https://nextpyp.app/files/pyp/latest/docs/install/troubleshooting.html#log-files), do you see any error messages in the logs (specifically the `micromon` one)?

**Reply by kexinliu0001** [reply]
hello, I met the same problem today.

I found that I cannot connect the localhost website, and tried to restart, and failed.
the logs in the micromon is:
<img width="1028" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/228944b0-40cd-4a52-b2fd-a55af74ce010">
<img width="1078" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/3a8ed4ff-9158-48a3-9c83-55180a4eb400">
and the latest update in the logs ([2024-01-09 14:07:55,676]) is 8 hours ago. I'm not sure if it's useful to know why nextpyp cannot be restarted now.
I tried to restart again, there is no error message updated in the micromon logs.

when systemctl status nextpyp:
<img width="1391" alt="image" src="https://github.com/nextpyp/.github/assets/110724494/ac5375c2-c129-46e2-819a-b15113360cfb">
it seemed to keep activating but cannot be activated and started successfully.
when journalctl -xe:
![image](https://github.com/nextpyp/.github/assets/110724494/382e2811-7b03-4288-8514-d2d2d91b08bd)

**Comment by cuchaz** [comment]
The relevant error message seems to be here:
```
A host processor is already running. Stop it before starting Micromon.
And then delete the files /tmp/micromon.hostprocessor.pid and /tmp/micromon.hostprocessor.pipe
```
Have you tried doing that and then restarting?

**Reply by kexinliu0001** [reply]
sorry, I still cannot find the error message you cited in the logs 😭

and do you mean it might be useful if:

systemctl stop nextPYP
delete the files in /tmp/micromon.hostprocessor.pid and /tmp/micromon.hostprocessor.pipe
and then
systemctl start nextPYP
?

**Reply by cuchaz** [reply, marked as answer]
If it helps, here's the error message you shared that I'm referring to:
![log](https://github.com/nextpyp/.github/assets/1892936/627db09f-0b0f-4286-962b-9b19261e0d9e)

Yes, try a small script like this:
```shell
sudo systemctl stop nextPYP
rm /tmp/micromon.hostprocessor.pid
rm /tmp/micromon.hostprocessor.pipe
sudo systemctl start nextPYP
```

Sometimes the shutdown process doesn't perform cleanup correctly and the next time the service is started, it looks like an old service is still running. That's where that error message comes from. If you know another service isn't actually running, then it's safe to do the cleanup manually (ie, delete the files) and then restart the service.

This seems to be a common problem with other people too, so we'll look into better ways to handle startup and shutdown. Thanks for reporting the issue!

**Reply by kexinliu0001** [reply]
Thank you! It really helps! 😄
